network/p2p/peerstore/utils.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package peerstore
+
+import (
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+// PeerInfoFromAddrs extracts the AddrInfo from a multiaddr string slice.
+func PeerInfoFromAddrs(addrs []string) ([]*peer.AddrInfo, error) {
+	var addrInfo []*peer.AddrInfo
+	for _, addr := range addrs {
+		info, err := PeerInfoFromAddr(addr)
+		if err != nil {
+			return nil, err
```

Comment:
So this will hard fail if a single input string of a potentially longish list is malformed. Is this the behavior that we want or should we log the error and continue?

The use-cases I'm thinking of here are if we use this codepath for `-p` option and phonebooks that users could manually edit. A typo could potentially prevent the node from accepting any of the inputs.

Reply:
good point. I will also return a list of bad addresses.